apply_DDT.py

```diff
@@ -0,0 +1,574 @@
+            X.append(icol.to_numpy(all_features))
+            weight.append(icol.xs / icol.cutflow['raw'] * lumi * icol.arrays['puweight'])
+        X = np.concatenate(X)
+        weight = np.concatenate(weight)
+
+    # If only one column
+    else: 
+        X = col.to_numpy(all_features)
+        weight = col.xs / col.cutflow['raw'] * lumi * col.arrays['puweight']
+  
+    # Apply the signal region
+    rt = X[:,-1]
+    X = X[rt > 1.18]
+    weight = weight[rt > 1.18]
+ 
+    # grab rt (after cut)
+    rt = rt[rt>1.18]
+    X = X[:,:-1]
+  
+    # grab rho
+    rho = X[:,-1]
+    X = X[:,:-1] # remove it from X so that eventually it can be used for BDT scores
+  
+    # grab mT
+    mT = X[:,-1]
+    X = X[:,:-1] # remove it from X so that eventually it can be used for BDT scores
+  
+    # grab pT
+    pT = X[:,-1]
+    X = X[:,:-1] # remove it from X so that eventually it can be used for BDT scores
+
+    return X, pT, mT, rho, weight
+
+
+#------------------------------------------------------------------------------
+# The Main Function -----------------------------------------------------------
+#------------------------------------------------------------------------------
+
+def main():
+    
+    # Parse arguments and take the results
+    args = parse_arguments()
+
+    bkg-files = args.bkg-files
+    sig-files = args.sig-files
+    model_file = args.bdt_file
+    ddt_map_file = args.ddt_map_file
+    lumi = args.lumi
+    sig_bdt_cut = args.sig_bdt_cut
+    bdt_cuts = args.bdt_cuts
+    plot_2D_DDT_map = args.plot_2D_DDT_map
+    plot_bkg_scores_mt = args.plot_bkg_scores_mt
+    plot_fom_significance = args.plot_fom_significance
+    plot_sig_mt_single_BDT = args.plot_sig_mt_single_BDT
+    plot_one_sig_mt_many_bdt = args.plot_one_sig_mt_many_bdt
+
+    set_mpl_fontsize(18, 22, 26)
+
+    # grab training features for specific input model
+    global training_features = read_training_features(model_file)
+
+    # add ddt necessary variables in addition to the bdt input features
+    all_features = training_features + ['pt', 'mt', 'rho', 'rt'] # rho is an important variable for applying the decorrelation
+
+    # Grab the bkg data
+    bkg_cols = [Columns.load(f) for f in glob.glob(bkg-files)]
+    #bkg_cols = [Columns.load(f) for f in glob.glob(qcd_data_files)]
+    #bkg_cols.extend(Columns.load(f) for f in glob.glob(tt_data_files))
+    X, pT, mT, rho, bkg_weight = bdt_ddt_inputs(bkg_cols, lumi, all_features)
+
+    # _____________________________________________
+    # Open the trained models and get the scores
+
+    bkg_score = {}
+
+    xgb_model = xgb.XGBClassifier()
+    xgb_model.load_model(model_file)
+    with time_and_log(f'Calculating xgboost scores for {model_file}...'):
+        bkg_score = xgb_model.predict_proba(X)[:,1]
+
+    # _____________________________________________
+    # Calculate bkg efficiencies for the DDT
+
+    print("Calculating bkg efficiencies for various bdt cut values")
```

Comment:
add a verbosity option to disable printouts

Reply:
Added